devito/finite_differences/derivative.py

```diff
@@ -206,7 +206,11 @@ def subs(self, *args, **kwargs):
         """
         Bypass sympy.Subs as Devito has its own lazy evaluation mechanism.
         """
-        return self.xreplace(dict(*args), **kwargs)
+        try:
+            rules = dict(*args)
```

Comment:
`rules = dict(as_tuple(args))` ?

Reply:
I wish but no doesn't work here. We have to case of input:
```
#dict
args = ({'a': x},)
# tuple
args = ('a', x)
```
as_tupe produce something `dict` doesn't like